File: typescript/packages/bench/src/rules/idleTechnicianRule.ts
/** Cardinality fixture: nominal (class, `implements`) implementor of RuleContract. */

import type { RuleContract } from "../contracts/ruleContract";
import type { Dataset } from "../dataset";

export class IdleTechnicianRule implements RuleContract {
    readonly key = "idle-technician";

    evaluate(data: Dataset): boolean {
        return data.technicians.length >= 1;
    }
}

File: rust/crates/atelier-bench/src/rules/idle_technician_rule.rs
//! Rule 24/24 (textual half).

use crate::contracts::rule_contract::RuleContract;
use crate::dataset::Dataset;

pub struct IdleTechnicianRule;

impl RuleContract for IdleTechnicianRule {
    fn evaluate(&self, data: &Dataset) -> bool {
        !data.technicians.is_empty()
    }
}
